`Error` is not `Send` anymore in v0.12.5

The last update breaks use cases like polling and parsing multiple feeds in parallel using threads.

In v0.12.4 code like this would compile just fine:

```rust
let handle = std::thread::spawn(
    || -> Result<atom_syndication::Feed, atom_syndication::Error> {
        // Poll feed and parse it.
        todo!()
    },
 );
```
With v0.12.5 compilation fails:

```
error[E0277]: `(dyn std::error::Error + 'static)` cannot be sent between threads safely
   --> poll/src/bin/test.rs:2:18
    |
2   |       let handle = std::thread::spawn(
    |  __________________^
3   | |         || -> Result<atom_syndication::Feed, atom_syndication::Error> {
4   | |             // Poll feed and parse it.
5   | |             todo!()
6   | |         },
7   | |     );
    | |_____^ `(dyn std::error::Error + 'static)` cannot be sent between threads safely
    |
    = help: the trait `Send` is not implemented for `(dyn std::error::Error + 'static)`, which is required by `Result<Feed, atom_syndication::Error>: Send`
    = note: required for `Unique<(dyn std::error::Error + 'static)>` to implement `Send`
```

The problem seems to be the change of `XmlError` from `pub struct XmlError(quick_xml::Error)` in v0.12.4 to `XmlError(Box<dyn StdError>)` in v0.12.5.
I don't know why that change was made, so can't suggest a fix for the described problem.

Many thanks for your work on this library.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+- Fix a regression: make `Error` and `XmlError` be `Send` and `Sync` again [`#89`](https://github.com/rust-syndication/atom/pull/89)
+
 ## 0.12.5 - 2024-11-16
 
 - Remove ambiguous statements about escaping from documentation. [`#85`](https://github.com/rust-syndication/atom/pull/85)

--- a/src/error.rs
+++ b/src/error.rs
@@ -64,10 +64,10 @@ impl From<XmlError> for Error {
 }
 
 #[derive(Debug)]
-pub struct XmlError(Box<dyn StdError>);
+pub struct XmlError(Box<dyn StdError + Send + Sync>);
 
 impl XmlError {
-    pub(crate) fn new(err: impl StdError + 'static) -> Self {
+    pub(crate) fn new(err: impl StdError + Send + Sync + 'static) -> Self {
         Self(Box::new(err))
     }
 }
@@ -81,5 +81,17 @@ impl StdError for XmlError {
 impl fmt::Display for XmlError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         fmt::Display::fmt(&self.0, f)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn error_send_and_sync() {
+        fn assert_send_sync<T: Send + Sync>() {}
+        assert_send_sync::<Error>();
+        assert_send_sync::<XmlError>();
     }
 }

